Add novu enterprise trail docker compose file

Cliftonz opened the pull request:
This add documentation and a docker compose file that only allows users to download the enterprise packages ONLY if they have a token from someone on the team with access to the novu packages.

Cliftonz committed:
Add Docker compose configuration for enterprise setup

This commit adds a new Docker Compose configuration for an enterprise setup. It also includes a corresponding README file to guide users through the process of setting up an on-premises trial of the enterprise solution using Docker Compose. This configuration currently supports multiple services like redis, mongodb, API server, worker, socket server, web app, widget, and embed server.

Cliftonz committed:
Update enterprise Readme with free trial note

A note has been added to the 'docker/enterprise/Readme.md' file to clarify that trying all features with a free trial on Novu Cloud is the preferred method, not the enterprise trail. This makes it clearer for users regarding the best way to try out the service.

Cliftonz committed:
Add warning note to enterprise Readme

A warning note has been added to the enterprise/Readme.md file in the 'Get the Code' section. This note informs users to remove any existing mongodb volume before starting the process to ensure the enterprise tables are installed correctly.

Cliftonz committed:
Update GH_TOKEN in enterprise Readme

The GH_TOKEN prefix used in the enterprise readme file has been updated from 'ghr' to 'ghp'. This change ensures that the correct token prefix is being used for authentication, aligning with the recent changes in the GitHub token system.

Cliftonz committed:
Add "servername" to .cspell.json

The "servername" has been appended to the .cspell.json configuration file for proper spell-checking and enhanced code quality.

denis-kralj-novu commented on docker/enterprise/Readme.md:
```diff
@@ -0,0 +1,65 @@
+
+This documentation guides you through the process of setting up an on-premises trial of Novu Enterprise using Docker Compose.
+
+To do this trail you will need to contact [sales](https://notify.novu.co/meetings/ryannovu/30minutes) to get a token.
```
trail -> trial

denis-kralj-novu commented on docker/enterprise/Readme.md:
```diff
@@ -0,0 +1,65 @@
+
+
+
+Note: This is not our preferred way of doing an enterprise trail,
```
trail -> trial

denis-kralj-novu commented on docker/enterprise/Readme.md:
```diff
@@ -0,0 +1,65 @@
+### Get the Code
+
+Warning: Before you start this process, If you have an existing mongodb volume you will need to remove it to 
+ensure the enterprise tables are installed correctly.
```
Is this an issue with the entire mongodb instance or a specific collection?

Cliftonz replied:
Yes, it creates different collections. I could not get my local deployment working with the db unless I removed everything that was existing.

denis-kralj-novu commented on docker/enterprise/Readme.md:
```diff
@@ -0,0 +1,65 @@
+cp .env.example ./enterprise/.env
+
+# Login
+echo "$GH_TOKEN" | docker login --username cliftonz --password-stdin
```
exchange specific username with a generic one

Cliftonz replied:
I cant the token is specifically tied to my account, but I can include that in the sales email instead of publicly.

Thanks!

denis-kralj-novu commented on docker/enterprise/docker-compose.enterprise.yml:
```diff
@@ -0,0 +1,185 @@
+    container_name: mongodb
+    restart: unless-stopped
+    logging:
+      driver: 'json-file'
```
how come we are interested in mongo's logs but not redis logs?

Cliftonz replied:
I would say to debug authentication but otherwise I am unsure

jainpawan21 commented on docker/enterprise/Readme.md:
````diff
@@ -0,0 +1,65 @@
+
+```sh
+# Add the token to the shell env
+GH_TOKEN=ghp_xxxx
````
A small suggestion to give this some novu specific name like

`NOVU_ENTERPRISE_TOKEN`

Shall we add this information in our docs as well?

Cliftonz committed:
Update enterprise trial documentation

Correct typos and rename GH_TOKEN to NOVU_ENTERPRISE_TOKEN in the Docker enterprise README file. Additionally, modify the docker login command to accommodate variable username. These changes improve readability and adaptability for users setting up an on-premises enterprise trial.

Cliftonz committed:
Merge branch 'next' into enterprise-demo
